<Type Name="DuplicateWaitObjectException" FullName="System.DuplicateWaitObjectException">
  <Metadata><Meta Name="ms.openlocfilehash" Value="ba5cbfa5f5819875b2b9bb0e112c0959fb19aaa0" /><Meta Name="ms.sourcegitcommit" Value="a3e01ec36374fac54a55add0dcc1e754e394abeb" /><Meta Name="ms.translationtype" Value="MT" /><Meta Name="ms.contentlocale" Value="ru-RU" /><Meta Name="ms.lasthandoff" Value="03/30/2019" /><Meta Name="ms.locfileid" Value="58685951" /></Metadata><TypeSignature Language="C#" Value="public class DuplicateWaitObjectException : ArgumentException" />
  <TypeSignature Language="ILAsm" Value=".class public auto ansi serializable beforefieldinit DuplicateWaitObjectException extends System.ArgumentException" />
  <TypeSignature Language="DocId" Value="T:System.DuplicateWaitObjectException" />
  <TypeSignature Language="VB.NET" Value="Public Class DuplicateWaitObjectException&#xA;Inherits ArgumentException" />
  <TypeSignature Language="C++ CLI" Value="public ref class DuplicateWaitObjectException : ArgumentException" />
  <TypeSignature Language="F#" Value="type DuplicateWaitObjectException = class&#xA;    inherit ArgumentException" />
  <AssemblyInfo>
    <AssemblyName>mscorlib</AssemblyName>
    <AssemblyVersion>1.0.5000.0</AssemblyVersion>
    <AssemblyVersion>2.0.0.0</AssemblyVersion>
    <AssemblyVersion>2.0.5.0</AssemblyVersion>
    <AssemblyVersion>4.0.0.0</AssemblyVersion>
  </AssemblyInfo>
  <AssemblyInfo>
    <AssemblyName>netstandard</AssemblyName>
    <AssemblyVersion>2.0.0.0</AssemblyVersion>
    <AssemblyVersion>2.1.0.0</AssemblyVersion>
  </AssemblyInfo>
  <AssemblyInfo>
    <AssemblyName>System.Runtime</AssemblyName>
    <AssemblyVersion>4.2.0.0</AssemblyVersion>
    <AssemblyVersion>4.2.1.0</AssemblyVersion>
  </AssemblyInfo>
  <Base>
    <BaseTypeName>System.ArgumentException</BaseTypeName>
  </Base>
  <Interfaces></Interfaces>
  <Attributes>
    <Attribute FrameworkAlternate="netframework-2.0;netframework-3.0;netframework-3.5;netframework-4.0;netframework-4.5;netframework-4.5.1;netframework-4.5.2;netframework-4.6;netframework-4.6.1;netframework-4.6.2;netframework-4.7;netframework-4.7.1;netframework-4.7.2;xamarinandroid-7.1;xamarinios-10.8;xamarinmac-3.0;netframework-4.8">
      <AttributeName>System.Runtime.InteropServices.ComVisible(true)</AttributeName>
    </Attribute>
    <Attribute FrameworkAlternate="netframework-1.1;netframework-2.0;netframework-3.0;netframework-3.5;netframework-4.0;netframework-4.5;netframework-4.5.1;netframework-4.5.2;netframework-4.6;netframework-4.6.1;netframework-4.6.2;netframework-4.7;netframework-4.7.1;netframework-4.7.2;netframework-4.8;xamarinandroid-7.1;xamarinios-10.8;xamarinmac-3.0">
      <AttributeName>System.Serializable</AttributeName>
    </Attribute>
  </Attributes>
  <Docs>
    <summary><span data-ttu-id="43756-101">Исключение, выбрасываемое, когда объект присутствует в массиве объектов синхронизации более одного раза.</span><span class="sxs-lookup"><span data-stu-id="43756-101">The exception that is thrown when an object appears more than once in an array of synchronization objects.</span></span></summary>
    <remarks>
      <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="43756-102">Среда CLR предоставляет механизм синхронизации потоков, на основе объектов синхронизации, ожидающих выполнения в массив <xref:System.Threading.WaitHandle> объектов.</span><span class="sxs-lookup"><span data-stu-id="43756-102">The common language runtime provides a thread synchronization mechanism based on synchronization objects waiting for execution in an array of <xref:System.Threading.WaitHandle> objects.</span></span> <span data-ttu-id="43756-103">Если массив <xref:System.Threading.WaitHandle> объекты передавать <xref:System.Threading.WaitHandle.WaitAll%2A> или <xref:System.Threading.WaitHandle.WaitAny%2A> содержит все повторяющиеся дескрипторы операционной системы, <xref:System.DuplicateWaitObjectException> возникает исключение.</span><span class="sxs-lookup"><span data-stu-id="43756-103">If the array of <xref:System.Threading.WaitHandle> objects passed to <xref:System.Threading.WaitHandle.WaitAll%2A> or <xref:System.Threading.WaitHandle.WaitAny%2A> contains any duplicate operating system handles, <xref:System.DuplicateWaitObjectException> is thrown.</span></span> <span data-ttu-id="43756-104">Дополнительные сведения см. в разделе <xref:System.Threading.WaitHandle>.</span><span class="sxs-lookup"><span data-stu-id="43756-104">For more information, see <xref:System.Threading.WaitHandle>.</span></span>  
  
 <span data-ttu-id="43756-105"><xref:System.DuplicateWaitObjectException> использует COR_E_DUPLICATEWAITOBJECT HRESULT, которое имеет значение 0x80131529.</span><span class="sxs-lookup"><span data-stu-id="43756-105"><xref:System.DuplicateWaitObjectException> uses the HRESULT COR_E_DUPLICATEWAITOBJECT, which has the value 0x80131529.</span></span>  
  
 <span data-ttu-id="43756-106">Список начальных значений свойств для экземпляра <xref:System.DuplicateWaitObjectException>, см. в разделе <xref:System.DuplicateWaitObjectException.%23ctor%2A> конструкторы.</span><span class="sxs-lookup"><span data-stu-id="43756-106">For a list of initial property values for an instance of <xref:System.DuplicateWaitObjectException>, see the <xref:System.DuplicateWaitObjectException.%23ctor%2A> constructors.</span></span>  
  
> [!NOTE]
>  <span data-ttu-id="43756-107">Это исключение не включается в [.NET для Windows Store apps](https://go.microsoft.com/fwlink/?LinkID=247912) или [переносимой библиотеки классов](~/docs/standard/cross-platform/cross-platform-development-with-the-portable-class-library.md), но он вызывается некоторые члены, которые являются.</span><span class="sxs-lookup"><span data-stu-id="43756-107">This exception is not included in [.NET for Windows Store apps](https://go.microsoft.com/fwlink/?LinkID=247912) or the [Portable Class Library](~/docs/standard/cross-platform/cross-platform-development-with-the-portable-class-library.md), but it is thrown by some members that are.</span></span> <span data-ttu-id="43756-108">Чтобы перехватить исключение в этом случае, написать `catch` инструкции для <xref:System.ArgumentException> вместо этого.</span><span class="sxs-lookup"><span data-stu-id="43756-108">To catch the exception in that case, write a `catch` statement for <xref:System.ArgumentException> instead.</span></span>  
  
 ]]></format>
    </remarks>
    <altmember cref="T:System.Threading.WaitHandle" />
    <altmember cref="T:System.Exception" />
    <related type="Article" href="~/docs/standard/threading/index.md"><span data-ttu-id="43756-109">Управляемая поточность</span><span class="sxs-lookup"><span data-stu-id="43756-109">Managed Threading</span></span></related>
    <related type="Article" href="~/docs/standard/exceptions/index.md"><span data-ttu-id="43756-110">Обработка и создание исключений</span><span class="sxs-lookup"><span data-stu-id="43756-110">Handling and Throwing Exceptions</span></span></related>
  </Docs>
  <Members>
    <MemberGroup MemberName=".ctor">
      <AssemblyInfo>
        <AssemblyName>mscorlib</AssemblyName>
        <AssemblyVersion>2.0.5.0</AssemblyVersion>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <Docs>
        <summary><span data-ttu-id="43756-111">Инициализирует новый экземпляр класса <see cref="T:System.DuplicateWaitObjectException" />.</span><span class="sxs-lookup"><span data-stu-id="43756-111">Initializes a new instance of the <see cref="T:System.DuplicateWaitObjectException" /> class.</span></span></summary>
      </Docs>
    </MemberGroup>
    <Member MemberName=".ctor">
      <MemberSignature Language="C#" Value="public DuplicateWaitObjectException ();" />
      <MemberSignature Language="ILAsm" Value=".method public hidebysig specialname rtspecialname instance void .ctor() cil managed" />
      <MemberSignature Language="DocId" Value="M:System.DuplicateWaitObjectException.#ctor" />
      <MemberSignature Language="VB.NET" Value="Public Sub New ()" />
      <MemberSignature Language="C++ CLI" Value="public:&#xA; DuplicateWaitObjectException();" />
      <MemberType>Constructor</MemberType>
      <AssemblyInfo>
        <AssemblyName>mscorlib</AssemblyName>
        <AssemblyVersion>1.0.5000.0</AssemblyVersion>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
        <AssemblyVersion>2.0.5.0</AssemblyVersion>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>netstandard</AssemblyName>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
        <AssemblyVersion>2.1.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>System.Runtime</AssemblyName>
        <AssemblyVersion>4.2.0.0</AssemblyVersion>
        <AssemblyVersion>4.2.1.0</AssemblyVersion>
      </AssemblyInfo>
      <Parameters />
      <Docs>
        <summary><span data-ttu-id="43756-112">Инициализирует новый экземпляр класса <see cref="T:System.DuplicateWaitObjectException" />.</span><span class="sxs-lookup"><span data-stu-id="43756-112">Initializes a new instance of the <see cref="T:System.DuplicateWaitObjectException" /> class.</span></span></summary>
        <remarks>
          <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="43756-113">Этот конструктор инициализирует <xref:System.Exception.Message%2A> свойства нового экземпляра системное сообщение с описанием ошибки, например «Повторяющиеся объекты в аргументе».</span><span class="sxs-lookup"><span data-stu-id="43756-113">This constructor initializes the <xref:System.Exception.Message%2A> property of the new instance to a system-supplied message that describes the error, such as "Duplicate objects in argument."</span></span> <span data-ttu-id="43756-114">Это сообщение учитывает текущую культуру системы.</span><span class="sxs-lookup"><span data-stu-id="43756-114">This message takes into account the current system culture.</span></span>  
  
 <span data-ttu-id="43756-115">В следующей таблице представлены исходные значения свойств экземпляра класса <xref:System.DuplicateWaitObjectException>.</span><span class="sxs-lookup"><span data-stu-id="43756-115">The following table shows the initial property values for an instance of <xref:System.DuplicateWaitObjectException>.</span></span>  
  
|<span data-ttu-id="43756-116">Свойство.</span><span class="sxs-lookup"><span data-stu-id="43756-116">Property</span></span>|<span data-ttu-id="43756-117">Значение</span><span class="sxs-lookup"><span data-stu-id="43756-117">Value</span></span>|  
|--------------|-----------|  
|<xref:System.Exception.InnerException%2A>|<span data-ttu-id="43756-118">Пустая ссылка (`Nothing` в Visual Basic).</span><span class="sxs-lookup"><span data-stu-id="43756-118">A null reference (`Nothing` in Visual Basic).</span></span>|  
|<xref:System.ArgumentException.Message%2A>|<span data-ttu-id="43756-119">Строка локализованного сообщения об ошибке.</span><span class="sxs-lookup"><span data-stu-id="43756-119">The localized error message string.</span></span>|  
  
 ]]></format>
        </remarks>
      </Docs>
    </Member>
    <Member MemberName=".ctor">
      <MemberSignature Language="C#" Value="public DuplicateWaitObjectException (string parameterName);" />
      <MemberSignature Language="ILAsm" Value=".method public hidebysig specialname rtspecialname instance void .ctor(string parameterName) cil managed" />
      <MemberSignature Language="DocId" Value="M:System.DuplicateWaitObjectException.#ctor(System.String)" />
      <MemberSignature Language="VB.NET" Value="Public Sub New (parameterName As String)" />
      <MemberSignature Language="C++ CLI" Value="public:&#xA; DuplicateWaitObjectException(System::String ^ parameterName);" />
      <MemberSignature Language="F#" Value="new DuplicateWaitObjectException : string -&gt; DuplicateWaitObjectException" Usage="new System.DuplicateWaitObjectException parameterName" />
      <MemberType>Constructor</MemberType>
      <AssemblyInfo>
        <AssemblyName>mscorlib</AssemblyName>
        <AssemblyVersion>1.0.5000.0</AssemblyVersion>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
        <AssemblyVersion>2.0.5.0</AssemblyVersion>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>netstandard</AssemblyName>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
        <AssemblyVersion>2.1.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>System.Runtime</AssemblyName>
        <AssemblyVersion>4.2.0.0</AssemblyVersion>
        <AssemblyVersion>4.2.1.0</AssemblyVersion>
      </AssemblyInfo>
      <Parameters>
        <Parameter Name="parameterName" Type="System.String" />
      </Parameters>
      <Docs>
        <param name="parameterName"><span data-ttu-id="43756-120">Имя параметра, вызвавшего данное исключение.</span><span class="sxs-lookup"><span data-stu-id="43756-120">The name of the parameter that caused the exception.</span></span></param>
        <summary><span data-ttu-id="43756-121">Инициализирует новый экземпляр класса <see cref="T:System.DuplicateWaitObjectException" /> именем параметра, вызвавшего исключение.</span><span class="sxs-lookup"><span data-stu-id="43756-121">Initializes a new instance of the <see cref="T:System.DuplicateWaitObjectException" /> class with the name of the parameter that causes this exception.</span></span></summary>
        <remarks>
          <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="43756-122">Содержимое параметра `parameterName` должно быть понятным пользователю.</span><span class="sxs-lookup"><span data-stu-id="43756-122">The content of `parameterName` is intended to be understood by humans.</span></span>  
  
 <span data-ttu-id="43756-123">В следующей таблице представлены исходные значения свойств экземпляра класса <xref:System.DuplicateWaitObjectException>.</span><span class="sxs-lookup"><span data-stu-id="43756-123">The following table shows the initial property values for an instance of <xref:System.DuplicateWaitObjectException>.</span></span>  
  
|<span data-ttu-id="43756-124">Свойство.</span><span class="sxs-lookup"><span data-stu-id="43756-124">Property</span></span>|<span data-ttu-id="43756-125">Значение</span><span class="sxs-lookup"><span data-stu-id="43756-125">Value</span></span>|  
|--------------|-----------|  
|<xref:System.Exception.InnerException%2A>|<span data-ttu-id="43756-126">Пустая ссылка (`Nothing` в Visual Basic).</span><span class="sxs-lookup"><span data-stu-id="43756-126">A null reference (`Nothing` in Visual Basic).</span></span>|  
|<xref:System.ArgumentException.Message%2A>|<span data-ttu-id="43756-127">Строка локализованного сообщения об ошибке.</span><span class="sxs-lookup"><span data-stu-id="43756-127">The localized error message string.</span></span>|  
|<xref:System.ArgumentException.ParamName%2A>|<span data-ttu-id="43756-128">Строка имени параметра.</span><span class="sxs-lookup"><span data-stu-id="43756-128">The parameter name string.</span></span>|  
  
 ]]></format>
        </remarks>
      </Docs>
    </Member>
    <Member MemberName=".ctor">
      <MemberSignature Language="C#" Value="protected DuplicateWaitObjectException (System.Runtime.Serialization.SerializationInfo info, System.Runtime.Serialization.StreamingContext context);" />
      <MemberSignature Language="ILAsm" Value=".method familyhidebysig specialname rtspecialname instance void .ctor(class System.Runtime.Serialization.SerializationInfo info, valuetype System.Runtime.Serialization.StreamingContext context) cil managed" />
      <MemberSignature Language="DocId" Value="M:System.DuplicateWaitObjectException.#ctor(System.Runtime.Serialization.SerializationInfo,System.Runtime.Serialization.StreamingContext)" />
      <MemberSignature Language="VB.NET" Value="Protected Sub New (info As SerializationInfo, context As StreamingContext)" />
      <MemberSignature Language="C++ CLI" Value="protected:&#xA; DuplicateWaitObjectException(System::Runtime::Serialization::SerializationInfo ^ info, System::Runtime::Serialization::StreamingContext context);" />
      <MemberSignature Language="F#" Value="new DuplicateWaitObjectException : System.Runtime.Serialization.SerializationInfo * System.Runtime.Serialization.StreamingContext -&gt; DuplicateWaitObjectException" Usage="new System.DuplicateWaitObjectException (info, context)" />
      <MemberType>Constructor</MemberType>
      <AssemblyInfo>
        <AssemblyName>mscorlib</AssemblyName>
        <AssemblyVersion>1.0.5000.0</AssemblyVersion>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
        <AssemblyVersion>2.0.5.0</AssemblyVersion>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>netstandard</AssemblyName>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
        <AssemblyVersion>2.1.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>System.Runtime</AssemblyName>
        <AssemblyVersion>4.2.0.0</AssemblyVersion>
        <AssemblyVersion>4.2.1.0</AssemblyVersion>
      </AssemblyInfo>
      <Attributes>
        <Attribute FrameworkAlternate="netframework-4.0">
          <AttributeName>System.Security.SecuritySafeCritical</AttributeName>
        </Attribute>
      </Attributes>
      <Parameters>
        <Parameter Name="info" Type="System.Runtime.Serialization.SerializationInfo" />
        <Parameter Name="context" Type="System.Runtime.Serialization.StreamingContext" />
      </Parameters>
      <Docs>
        <param name="info"><span data-ttu-id="43756-129">Объект, содержащий сериализованные данные объекта.</span><span class="sxs-lookup"><span data-stu-id="43756-129">The object that holds the serialized object data.</span></span></param>
        <param name="context"><span data-ttu-id="43756-130">Контекстные сведения об источнике или назначении.</span><span class="sxs-lookup"><span data-stu-id="43756-130">The contextual information about the source or destination.</span></span></param>
        <summary><span data-ttu-id="43756-131">Инициализирует новый экземпляр класса <see cref="T:System.DuplicateWaitObjectException" /> с сериализованными данными.</span><span class="sxs-lookup"><span data-stu-id="43756-131">Initializes a new instance of the <see cref="T:System.DuplicateWaitObjectException" /> class with serialized data.</span></span></summary>
        <remarks>
          <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="43756-132">Данный конструктор вызывается в процессе десериализации для воссоздания объекта исключения, переданного в потоке.</span><span class="sxs-lookup"><span data-stu-id="43756-132">This constructor is called during deserialization to reconstitute the exception object transmitted over a stream.</span></span> <span data-ttu-id="43756-133">Дополнительные сведения см. в разделе [сериализация XML и SOAP](~/docs/standard/serialization/xml-and-soap-serialization.md).</span><span class="sxs-lookup"><span data-stu-id="43756-133">For more information, see [XML and SOAP Serialization](~/docs/standard/serialization/xml-and-soap-serialization.md).</span></span>  
  
 ]]></format>
        </remarks>
        <related type="Article" href="~/docs/standard/serialization/xml-and-soap-serialization.md"><span data-ttu-id="43756-134">Сериализация XML и SOAP</span><span class="sxs-lookup"><span data-stu-id="43756-134">XML and SOAP Serialization</span></span></related>
      </Docs>
    </Member>
    <Member MemberName=".ctor">
      <MemberSignature Language="C#" Value="public DuplicateWaitObjectException (string message, Exception innerException);" />
      <MemberSignature Language="ILAsm" Value=".method public hidebysig specialname rtspecialname instance void .ctor(string message, class System.Exception innerException) cil managed" />
      <MemberSignature Language="DocId" Value="M:System.DuplicateWaitObjectException.#ctor(System.String,System.Exception)" />
      <MemberSignature Language="VB.NET" Value="Public Sub New (message As String, innerException As Exception)" />
      <MemberSignature Language="C++ CLI" Value="public:&#xA; DuplicateWaitObjectException(System::String ^ message, Exception ^ innerException);" />
      <MemberSignature Language="F#" Value="new DuplicateWaitObjectException : string * Exception -&gt; DuplicateWaitObjectException" Usage="new System.DuplicateWaitObjectException (message, innerException)" />
      <MemberType>Constructor</MemberType>
      <AssemblyInfo>
        <AssemblyName>mscorlib</AssemblyName>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
        <AssemblyVersion>2.0.5.0</AssemblyVersion>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>netstandard</AssemblyName>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
        <AssemblyVersion>2.1.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>System.Runtime</AssemblyName>
        <AssemblyVersion>4.2.0.0</AssemblyVersion>
        <AssemblyVersion>4.2.1.0</AssemblyVersion>
      </AssemblyInfo>
      <Parameters>
        <Parameter Name="message" Type="System.String" Index="0" FrameworkAlternate="netcore-2.0;netcore-2.1;netcore-2.2;netcore-3.0;netframework-2.0;netframework-3.0;netframework-3.5;netframework-4.0;netframework-4.5;netframework-4.5.1;netframework-4.5.2;netframework-4.6;netframework-4.6.1;netframework-4.6.2;netframework-4.7;netframework-4.7.1;netframework-4.7.2;netframework-4.8;netstandard-2.0;netstandard-2.1;xamarinandroid-7.1;xamarinios-10.8;xamarinmac-3.0" />
        <Parameter Name="innerException" Type="System.Exception" Index="1" FrameworkAlternate="netcore-2.0;netcore-2.1;netcore-2.2;netcore-3.0;netframework-2.0;netframework-3.0;netframework-3.5;netframework-4.0;netframework-4.5;netframework-4.5.1;netframework-4.5.2;netframework-4.6;netframework-4.6.1;netframework-4.6.2;netframework-4.7;netframework-4.7.1;netframework-4.7.2;netframework-4.8;netstandard-2.0;netstandard-2.1;xamarinandroid-7.1;xamarinios-10.8;xamarinmac-3.0" />
      </Parameters>
      <Docs>
        <param name="message"><span data-ttu-id="43756-135">Сообщение об ошибке, указывающее причину создания исключения.</span><span class="sxs-lookup"><span data-stu-id="43756-135">The error message that explains the reason for the exception.</span></span></param>
        <param name="innerException"><span data-ttu-id="43756-136">Исключение, которое является причиной текущего исключения.</span><span class="sxs-lookup"><span data-stu-id="43756-136">The exception that is the cause of the current exception.</span></span> <span data-ttu-id="43756-137">Если значение параметра <paramref name="innerException" /> не равно <see langword="null" />, текущее исключение сгенерировано в блоке <see langword="catch" />, обрабатывающем внутреннее исключение.</span><span class="sxs-lookup"><span data-stu-id="43756-137">If the <paramref name="innerException" /> parameter is not <see langword="null" />, the current exception is raised in a <see langword="catch" /> block that handles the inner exception.</span></span></param>
        <summary><span data-ttu-id="43756-138">Инициализирует новый экземпляр класса <see cref="T:System.DuplicateWaitObjectException" /> указанным сообщением об ошибке и ссылкой на внутреннее исключение, вызвавшее данное исключение.</span><span class="sxs-lookup"><span data-stu-id="43756-138">Initializes a new instance of the <see cref="T:System.DuplicateWaitObjectException" /> class with a specified error message and a reference to the inner exception that is the cause of this exception.</span></span></summary>
        <remarks>
          <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="43756-139">Исключение, созданное как прямой результат предыдущего исключения, должно содержать в свойстве <xref:System.Exception.InnerException%2A> ссылку на предыдущее исключение.</span><span class="sxs-lookup"><span data-stu-id="43756-139">An exception that is thrown as a direct result of a previous exception should include a reference to the previous exception in the <xref:System.Exception.InnerException%2A> property.</span></span> <span data-ttu-id="43756-140"><xref:System.Exception.InnerException%2A> Свойство возвращает то же значение, который передается этому конструктору.</span><span class="sxs-lookup"><span data-stu-id="43756-140">The <xref:System.Exception.InnerException%2A> property returns the same value that is passed to this constructor.</span></span>  
  
 <span data-ttu-id="43756-141">В следующей таблице представлены исходные значения свойств экземпляра класса <xref:System.DuplicateWaitObjectException>.</span><span class="sxs-lookup"><span data-stu-id="43756-141">The following table shows the initial property values for an instance of <xref:System.DuplicateWaitObjectException>.</span></span>  
  
|<span data-ttu-id="43756-142">Свойство.</span><span class="sxs-lookup"><span data-stu-id="43756-142">Property</span></span>|<span data-ttu-id="43756-143">Значение</span><span class="sxs-lookup"><span data-stu-id="43756-143">Value</span></span>|  
|--------------|-----------|  
|<xref:System.Exception.InnerException%2A>|<span data-ttu-id="43756-144">Исключение, заданное параметром `innerException`.</span><span class="sxs-lookup"><span data-stu-id="43756-144">The exception specified by `innerException`.</span></span>|  
|<xref:System.ArgumentException.Message%2A>|<span data-ttu-id="43756-145">Строка сообщения об ошибке определяется `message`.</span><span class="sxs-lookup"><span data-stu-id="43756-145">The error message string specified by `message`.</span></span>|  
  
 ]]></format>
        </remarks>
      </Docs>
    </Member>
    <Member MemberName=".ctor">
      <MemberSignature Language="C#" Value="public DuplicateWaitObjectException (string parameterName, string message);" />
      <MemberSignature Language="ILAsm" Value=".method public hidebysig specialname rtspecialname instance void .ctor(string parameterName, string message) cil managed" />
      <MemberSignature Language="DocId" Value="M:System.DuplicateWaitObjectException.#ctor(System.String,System.String)" />
      <MemberSignature Language="VB.NET" Value="Public Sub New (parameterName As String, message As String)" />
      <MemberSignature Language="C++ CLI" Value="public:&#xA; DuplicateWaitObjectException(System::String ^ parameterName, System::String ^ message);" />
      <MemberSignature Language="F#" Value="new DuplicateWaitObjectException : string * string -&gt; DuplicateWaitObjectException" Usage="new System.DuplicateWaitObjectException (parameterName, message)" />
      <MemberType>Constructor</MemberType>
      <AssemblyInfo>
        <AssemblyName>mscorlib</AssemblyName>
        <AssemblyVersion>1.0.5000.0</AssemblyVersion>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
        <AssemblyVersion>2.0.5.0</AssemblyVersion>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>netstandard</AssemblyName>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
        <AssemblyVersion>2.1.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>System.Runtime</AssemblyName>
        <AssemblyVersion>4.2.0.0</AssemblyVersion>
        <AssemblyVersion>4.2.1.0</AssemblyVersion>
      </AssemblyInfo>
      <Parameters>
        <Parameter Name="parameterName" Type="System.String" />
        <Parameter Name="message" Type="System.String" />
      </Parameters>
      <Docs>
        <param name="parameterName"><span data-ttu-id="43756-146">Имя параметра, вызвавшего данное исключение.</span><span class="sxs-lookup"><span data-stu-id="43756-146">The name of the parameter that caused the exception.</span></span></param>
        <param name="message"><span data-ttu-id="43756-147">Сообщение, описывающее ошибку.</span><span class="sxs-lookup"><span data-stu-id="43756-147">The message that describes the error.</span></span></param>
        <summary><span data-ttu-id="43756-148">Инициализирует новый экземпляр класса <see cref="T:System.DuplicateWaitObjectException" /> с указанным сообщением об ошибке и именем параметра, ставшего причиной этого исключения.</span><span class="sxs-lookup"><span data-stu-id="43756-148">Initializes a new instance of the <see cref="T:System.DuplicateWaitObjectException" /> class with a specified error message and the name of the parameter that causes this exception.</span></span></summary>
        <remarks>
          <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="43756-149">Содержание `parameterName` и `message` должно быть понятным пользователю.</span><span class="sxs-lookup"><span data-stu-id="43756-149">The content of the `parameterName` and `message` parameters is intended to be understood by humans.</span></span> <span data-ttu-id="43756-150">Код, вызывающий этот конструктор необходим, чтобы убедиться, что эти строки локализованные для текущего языка и региональных параметров системы.</span><span class="sxs-lookup"><span data-stu-id="43756-150">The caller of this constructor is required to ensure that these strings have been localized for the current system culture.</span></span>  
  
 <span data-ttu-id="43756-151">Исключение, созданное как прямой результат предыдущего исключения, должно содержать в свойстве <xref:System.Exception.InnerException%2A> ссылку на предыдущее исключение.</span><span class="sxs-lookup"><span data-stu-id="43756-151">An exception that is thrown as a direct result of a previous exception should include a reference to the previous exception in the <xref:System.Exception.InnerException%2A> property.</span></span> <span data-ttu-id="43756-152">Свойство <xref:System.Exception.InnerException%2A> возвращает то же значение, которое передается конструктору, или пустую ссылку (`Nothing` в Visual Basic), если свойство <xref:System.Exception.InnerException%2A> не предоставляет конструктору значение внутреннего исключения.</span><span class="sxs-lookup"><span data-stu-id="43756-152">The <xref:System.Exception.InnerException%2A> property returns the same value that is passed into the constructor, or a null reference (`Nothing` in Visual Basic) if the <xref:System.Exception.InnerException%2A> property does not supply the inner exception value to the constructor.</span></span>  
  
 <span data-ttu-id="43756-153">В следующей таблице представлены исходные значения свойств экземпляра класса <xref:System.DuplicateWaitObjectException>.</span><span class="sxs-lookup"><span data-stu-id="43756-153">The following table shows the initial property values for an instance of <xref:System.DuplicateWaitObjectException>.</span></span>  
  
|<span data-ttu-id="43756-154">Свойство.</span><span class="sxs-lookup"><span data-stu-id="43756-154">Property</span></span>|<span data-ttu-id="43756-155">Значение</span><span class="sxs-lookup"><span data-stu-id="43756-155">Value</span></span>|  
|--------------|-----------|  
|<xref:System.Exception.InnerException%2A>|<span data-ttu-id="43756-156">Пустая ссылка (`Nothing` в Visual Basic)</span><span class="sxs-lookup"><span data-stu-id="43756-156">A null reference (`Nothing` in Visual Basic)</span></span>|  
|<xref:System.ArgumentException.Message%2A>|<span data-ttu-id="43756-157">Строка сообщения об ошибке.</span><span class="sxs-lookup"><span data-stu-id="43756-157">The error message string.</span></span>|  
|<xref:System.ArgumentException.ParamName%2A>|<span data-ttu-id="43756-158">Строка имени параметра.</span><span class="sxs-lookup"><span data-stu-id="43756-158">The parameter name string.</span></span>|  
  
 ]]></format>
        </remarks>
        <altmember cref="T:System.Exception" />
        <related type="Article" href="~/docs/standard/exceptions/index.md"><span data-ttu-id="43756-159">Обработка и создание исключений</span><span class="sxs-lookup"><span data-stu-id="43756-159">Handling and Throwing Exceptions</span></span></related>
      </Docs>
    </Member>
  </Members>
</Type>